### PR TITLE
Check failure store when available during system tests

### PR DESCRIPTION
**Pull request by jsoriano**
Enable failure store and check if it has any documents after executing system tests.

Failure store is used if the version of the stack is 8.15.0 or newer. In lower versions there are unexpected failures on tests.

Fixes https://github.com/elastic/elastic-package/issues/1832.

### CODE REVIEW

**Comment by jsoriano** [comment]
test integrations

**Comment by jsoriano** [comment]
test integrations

**Comment by jsoriano** [comment]
test integrations

File: cmd/testrunner.go [hunk 1]
```diff
@@ -534,6 +535,15 @@ func testRunnerSystemCommandAction(cmd *cobra.Command, args []string) error {
...
 	}
+	versionInfo, err := kibanaClient.Version()
+	if err != nil {
+		return fmt.Errorf("can't get version info from Kibana client: %w", err)
+	}
+	stackVersion, err := semver.NewVersion(versionInfo.Number)
+	if err != nil {
+		return fmt.Errorf("can't parse Kibana version %q: %w", versionInfo.Number, err)
+	}
+	checkFailureStore := !stackVersion.LessThan(semver.MustParse("8.14.0"))
```
Comment (jsoriano) [inline cmd/testrunner.go:546]: We should not rely only on the version, but actually check that ES supports the failure store.

**Comment by jsoriano** [comment]
test integrations

**Comment by jsoriano** [comment]
Ok, it looks like even when the failure store is available in older versions, it has some issues with time series before 8.15. Reopening for review after enabling failure store starting on 8.15, and pinning the stack version for the related test package.

**Comment by jsoriano** [comment]
test integrations

**Review by jlind23 — APPROVED** [review]
LGTM but waiting for @flash1293  review and test anyway

**Comment by flash1293** [comment]
The code looks good to me, however I'm not sure how to test locally anymore - I used to 

`elastic-package test system -v --setup --config-file ./my/config.yml`, then `elastic-package test system -v --no-provision`.

The first command still works and finds the docs in the failure store, however I don't know how to see the test failing locally (as the setup routine is not running the validation bits). Seems like this code was recently restructured.

Not a blocker to merge though, I read through the code and it seems to work exactly like the ignored docs stuff.

**Comment by jsoriano** [comment]
@flash1293 do you mean that it doesn't fail when running with `--no-provision`? It should fail the same as when running directly with `elastic-package test system -v`.

**Comment by flash1293** [comment]
It never finds any hits when running with --no-provision and doesn't terminate

**Comment by jsoriano** [comment]
> It never finds any hits when running with --no-provision and doesn't terminate

Have you tried with `elastic-package test system -v`? Could you share the package you are testing with?

**Comment by flash1293** [comment]
@jsoriano When I'm running just with `test system` it doesn't get to that point because it fails during teardown:
> Error: failed to tear down system runner: there was an apply error: can't remove the package: could not remove package; API status code = 400; response body = {"statusCode":400,"error":"Bad Request","message":"Unable to remove package with existing package policy(s) in use by agent(s)"}

I'm testing with the `test/packages/false_positives/failure_store` package from this PR.

I don't think it's something wrong with this PR, seems like a general thing.

**Review by flash1293 — APPROVED** [review]
nvm, I think it was just some stale state in my local stack. Tested again with a fresh stack and it works as expected 👍 

One nit: Maybe we should show more details of the failure than just the error message (e.g. which processor was involved and so on) to make it easier to debug what's happening.

**Comment by jsoriano** [comment]
> Maybe we should show more details of the failure than just the error message (e.g. which processor was involved and so on) to make it easier to debug what's happening.

Thanks @flash1293 for the feedback. I have pushed another commit including more information in the error messages, they include the error type, the processor and the pipelines that caused the issue:
```
--- Test results for package: failure_store - START ---
FAILURE DETAILS:
failure_store/test fail:
[0] illegal_argument_exception: unable to convert [eight] to integer (processor: convert, pipelines: logs-failure_store.test-0.0.1)
[1] illegal_argument_exception: unable to convert [five] to integer (processor: convert, pipelines: logs-failure_store.test-0.0.1)
[2] illegal_argument_exception: unable to convert [seven] to integer (processor: convert, pipelines: logs-failure_store.test-0.0.1)
[3] illegal_argument_exception: unable to convert [six] to integer (processor: convert, pipelines: logs-failure_store.test-0.0.1)
```

Additionally, the stack trace is logged at the debug level when collecting results.

Wdyt? Is it too much?

**Comment by flash1293** [comment]
Love it! Thanks for adding

**Review by flash1293 — APPROVED** [review]

**Comment by jsoriano** [comment]
test integrations

**Pull request merged**